subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/scheduler/TestSchedulerFactoryImpl.kt

```diff
@@ -1,47 +1,229 @@
+
+        val metricsConfig: RunnerMetricsConfig = createRunnerMetricsConfig(params)
+
+        val finalizerFactory: FinalizerFactory = createFinalizerFactory(
+            params = params,
+            metricsConfig = metricsConfig,
+            reportFactory = reportFactory,
+            timeProvider = timeProvider
+        )
+
+        val androidDebugBridgeProvider = createAndroidDebugBridgeProvider(params.loggerFactory)
+
+        val emulatorsLogsReporterProvider = createEmulatorsLogsReporterProvider(params)
+
+        val testRunnerFactoryProvider = createTestRunnerFactoryProvider(
+            params = params,
+            timeProvider = timeProvider,
+            httpClientProvider = httpClientProvider,
+            report = report,
+            metricsConfig = metricsConfig,
+            androidDebugBridgeProvider = androidDebugBridgeProvider,
+            emulatorsLogsReporterProvider = emulatorsLogsReporterProvider
+        )
 
         return TestSchedulerImpl(
             params = params,
             report = report,
             testSuiteProvider = testSuiteProvider,
-            testSuiteLoader = testSuiteLoader,
+            testSuiteLoader = createTestSuiteLoader(),
             filterInfoWriter = FilterInfoWriter.Impl(
                 outputDir = params.outputDir,
             ),
             loggerFactory = params.loggerFactory,
             testRunnerFactory = testRunnerFactoryProvider.provide(),
+            finalizer = finalizerFactory.create(),
         )
     }
 
-    private fun createTestSuiteProvider(): TestSuiteProvider = TestSuiteProvider.Impl(
-        report = report,
-        targets = params.instrumentationConfiguration.targets,
-        reportSkippedTests = params.instrumentationConfiguration.reportSkippedTests,
-        filterFactory = FilterFactory.create(
+    private fun createFinalizerFactory(
+        params: RunnerInputParams,
+        metricsConfig: RunnerMetricsConfig,
+        reportFactory: ReportFactory,
+        timeProvider: TimeProvider,
+    ): FinalizerFactory {
+        return FinalizerFactoryImpl(
+            params = params,
+            metricsConfig = metricsConfig,
+            reportFactory = reportFactory,
+            timeProvider = timeProvider,
+            loggerFactory = params.loggerFactory,
+        )
+    }
+
+    private fun createRunnerMetricsConfig(params: RunnerInputParams): RunnerMetricsConfig {
+        return RunnerMetricsConfig(
+            statsDConfig = params.statsDConfig,
+            runnerPrefix = SeriesName.create(
+                "testrunner",
+                params.projectName,
+                params.instrumentationConfiguration.name
+            )
+        )
+    }
+
+    private fun createStatsdSender(params: RunnerInputParams): StatsDSender {
+        return StatsDSender.Impl(
+            config = params.statsDConfig,
+            loggerFactory = params.loggerFactory
+        )
+    }
+
+    private fun createTestSuiteLoader(): TestSuiteLoader {
```

Comment:
Useless

Reply:
hides implementation, i had plans to look why Implementation class is here, and not hidden in some module

Reply:
I don't want to debate lets discuss it at the call